runtime/gc_base/modronapi.cpp

```diff
@@ -65,6 +70,12 @@ postDismountContinuation(J9VMThread *vmThread, j9object_t object)
 	/* Conservatively assume that via mutations of stack slots (which are not subject to access barriers),
 	 * all post-write barriers have been triggered on this Continuation object, since it's been mounted. */
 	vmThread->javaVM->memoryManagerFunctions->J9WriteBarrierBatch(vmThread, object);
+#if JAVA_SPEC_VERSION >= 19
+		J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(vmThread, object);
```

Comment:
Currious, under which circumstances, this pointer could be NULL?

Reply:
for this case continuation could not be NULL, change it as `Assert_MM_true(NULL != continuation);`

Reply:
On the VM side, `JVM_VirtualThreadUnmountEnd` will set it to NULL during the last unmount of a virtual thread. During the last unmount, the virtual thread state is set to `TERMINATED`. RI collects such virtual threads. RI also collects virtual threads if their state is `NEW`. Currently, we set `CONTINUATION_VMREF` for a virtual thread from its constructor when its state is `NEW`. Will this prevent a virtual thread in state NEW to be collected? If so, `CONTINUATION_VMREF` should be set during the first `Cont.enterImpl`.

Reply:
iirc, the `vmRef` field of a continuation is set during constructor code (`createContinuation`), so even if a new continuation that never gets executed will still have the field set.